### PR TITLE
ShowMore in redux state

### PR DESCRIPTION
**Pull request by apolignano**

### CODE REVIEW

**Comment by apolignano** [comment]
- The ShowMore button state is now saved in the Global Redux State, allowing users to keep the preferred view.
- The styling of the toggle button could be up for discussion to make it as similar as possible to the other VF components.

**Review by matthiasblum — CHANGES_REQUESTED** [review]
The switch works as intended, but there are a few things that could be improved

1. it looks like you have re-implemented a switch button. Can you use the [`ToggleSwitch`](https://github.com/ProteinsWebTeam/interpro7-client/blob/master/src/components/ToggleSwitch/index.tsx) component instead?
2. The More/Less option isn't so clear. We could add a label _Feature display_ in front of the switch and the options of the switch would be _Overview_ and _Full_.

**Comment by apolignano** [comment]
Unfortunately, the ToggleSwitch component doesn't look great when positioned there, as it only accepts three sizes and none of them match the Option menu button. I've tried to pass an height prop but you end up having to play around too much with margins, CSS vars, etc.. Tried the label too, and it looks even worse IMO. 

Less and More are renamed to Overview and Full and the color-changing behavior of ToggleSwitch has been implemented.

Screenshot 2024-11-22 at 12 26 40

Screenshot 2024-11-22 at 12 28 09

Let me know if it works!

File: src/components/ProteinViewer/ShowMoreTracks/index.tsx [hunk 1]
```diff
@@ -20,44 +19,47 @@ type Props = {
...
-      </button>
+        <div
+          className={css('showmore-toggle-inner', showMore ? 'moved' : '')}
+        />
+        <span style={{ marginLeft: showMore ? '7px' : '25px' }}>
+          {showMore ? 'Full View' : 'Overview'}
+        </span>
+      </div>
     </div>
   );
```
Comment (matthiasblum) [inline src/components/ProteinViewer/ShowMoreTracks/index.tsx:64]: Minor style changes, and add a label in front of the switch (not the best UI, but I think we need a label):

```ts
  return (
    <>
      Feature display:
      <div onClick={handleClick}>
        <div
          className={css('showmore-toggle-outer', showMore ? 'full' : 'overview')}
        >
          <div
            className={css('showmore-toggle-inner', showMore ? 'moved' : '')}
          />
          <span style={{ marginLeft: showMore ? '5px' : '33px' }}>
          {showMore ? 'Full' : 'Overview'}
        </span>
        </div>
      </div>
    </>
  );
```

File: src/components/ProteinViewer/ShowMoreTracks/style.css [hunk 2]
```diff
@@ -1,3 +1,39 @@
 .showmore-btn {
   margin: 0;
 }
+
+.showmore-toggle-outer {
+  display: flex;
+  align-items: center;
+  width: 105px;
```
Comment (matthiasblum) [inline src/components/ProteinViewer/ShowMoreTracks/style.css:8]:
```suggestion
  width: 118px;
```

File: src/components/ProteinViewer/ShowMoreTracks/style.css [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+
+.showmore-toggle-outer.full {
+  background-color: #1779ba;
+}
+
+.showmore-toggle-inner {
+  position: absolute;
+  top: 5px;
+  left: 5px;
+  width: 15px;
```
Comment (matthiasblum) [inline src/components/ProteinViewer/ShowMoreTracks/style.css:31]:
```suggestion
  width: 28px;
```

**Review by matthiasblum — COMMENTED** [review]
Looks better. I suggested a few changes that I think could improve the UI. Please have a look and let me know if you agree with them.

**Pull request merged**